index.js

```diff
@@ -0,0 +1,295 @@
+      }
+    });
+};
+
+const buildForm = function buildForm(id, name) {
+  let formContent = '';
+  formContent += `<h3>Reserve Trip: ${name}</h3>`;
+  formContent += `<form action=${baseURL}>`;
+  formContent += '<div><label for="name">Name&emsp;</label> <input type="text" name="name" /></div>';
+  formContent += '<div><label for="email">Email&emsp;</label> <input type="text" name="email" /></div>';
+  formContent += `<input type='hidden' name='id' value=${id}>`;
+  formContent += '<button type="submit" class="button">Reseve Trip!</button>';
+
+  $('#trip-form').html(formContent);
+};
+
+const getTripInfo = function getTripInfo(event) {
+  let tripInfo = $('#trip-info');
+  tripInfo.empty();
+
+  let target = event.target;
+  let tripLink = baseURL + '/' + target.id.toString();
+
+  axios.get(tripLink)
+    .then( (response) => {
+        let trip = response.data;
+        let content = '<table>';
+        content += `<tr> <td>ID</td> <td> ${trip.id} </td></tr>`;
+        content += `<tr> <td>NAME</td> <td> ${trip.name} </td></tr>`;
+        content += `<tr> <td>COST</td> <td> $${trip.cost} </td></tr>`;
+        content += `<tr> <td>WEEKS</td> <td>${trip.weeks}</td> </tr>`
+        content += `<tr> <td>CATEGORY</td> <td> ${trip.category} </td></tr>`;
+        content += `<tr> <td id='about-trip'>ABOUT</td> <td> ${trip.about} </td></tr>`;
+        content += '</table>';
+        $('#trip-info').append('<h3>Trip Info</h3>');
+        $('#trip-info').append(content);
+        buildForm(trip.id, trip.name);
+    })
+    .catch( (error) => {
+      if (error.response.data && error.response.data.errors) {
+        reportError(
+          `Encountered an error: ${error.message}`,
+          error.response.data.errors
+        );
+      } else {
+        reportStatus(`Error: ${error.message}`, 'failure');
+      }
+    });
+
+};
+
+const reserveTrip = function reserveTrip (event) {
+
+  event.preventDefault();
+
+  const reservationURL = baseURL + '/' + $('input[name="id"]').val() + '/reservations'
+
+  const tripData = {
+    name: $('input[name="name"]').val(),
+    email: $('input[name="email"]').val(),
+  }
+
+  reportStatus('Attempting to book trip...');
+
+  axios.post(reservationURL, tripData)
+    .then(() => {
+      $('input[name="name"]').val(''); // clearing text field
+      $('input[name="email"]').val('');
+
+      reportStatus('Successfully booked trip!', 'success');
+    })
+    .catch((error) => {
+      if (error.response.data && error.response.data.errors) {
+        reportError(
+          `Encountered an error:`,
+          error.response.data.errors
+        );
+      } else {
+        reportStatus(`Error: ${error.message}`, 'failure');
+      }
+    });
+
+};
+
+
+// QUERY SEARCHES
+const searchBySection = function searchBy (event) {
+  let target = event.target;
+  let searchCat = target.id;
+
+  switch(searchCat) {
+    case 'continent':
+      $('#search-filters').html(
+        `<div class='dropdown'>
```

Comment:
I'd love to see each of these dropdowns returned from separate functions to clean up this function